examples/controls_example/public/react_controls/initialize_default_control_api.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { BehaviorSubject } from 'rxjs';
+
+import { ControlWidth } from '@kbn/controls-plugin/common';
+import { SerializedPanelState } from '@kbn/presentation-containers';
+import { StateComparators } from '@kbn/presentation-publishing';
+
+import {
+  ControlApiInitialization,
+  ControlStateManager,
+  DefaultControlApi,
+  DefaultControlState,
+} from './types';
+
+export type ControlApi = ControlApiInitialization<DefaultControlApi>;
+
+export const initializeDefaultControlApi = (
+  state: DefaultControlState
+): {
+  defaultControlApi: ControlApi;
+  defaultControlStateManager: ControlStateManager<DefaultControlState>;
+  defaultControlComparators: StateComparators<DefaultControlState>;
+  serializeDefaultControl: () => SerializedPanelState<DefaultControlState>;
+} => {
+  const dataLoading = new BehaviorSubject<boolean | undefined>(false);
+  const blockingError = new BehaviorSubject<Error | undefined>(undefined);
+  const grow = new BehaviorSubject<boolean | undefined>(state.grow);
+  const width = new BehaviorSubject<ControlWidth | undefined>(state.width);
+
+  const defaultControlApi: ControlApi = {
+    grow,
+    width,
+    dataLoading,
+    blockingError,
+    setBlockingError: (error) => blockingError.next(error),
+    setDataLoading: (loading) => dataLoading.next(loading),
```

Comment:
I need to make setters for loading and error so that each control that uses either `initializeDefaultControlApi` or `initializeDataControl` can set these values - not sure if there is a better way to handle this? I could include `dataLoading` and `blockingError` in the state manager, but they don't really belong under "state" to me 🤔